include/utils/error.hpp

```diff
@@ -24,13 +24,15 @@ RequestClearErrorOption string_to_request_clear_error_option(const std::string&
 
 class NotValidErrorType : public std::exception {
 public:
-    explicit NotValidErrorType(const std::string& error_type_) : error_type(error_type_){};
-    virtual const char* what() const throw() {
-        return fmt::format("Error type '{}' is not valid.", error_type).c_str();
+    explicit NotValidErrorType(const std::string& error_type_) :
+        error_type(error_type_), explanatory_string(fmt::format("Error type '{}' is not valid.", error_type)){};
```

Comment:
Is there any reason, why `error_type` should be kept as a member variable?

Reply:
@andistorm ?

Reply:
`error_type` as member variable might allows to handle the exception in specific cases. Right now there is no usage of that. And it should be probably public to allow using it..